File: Problemas/1915B - Not Quite Latin Square/1915B_Not_Quite_Latin_Square.py
# 1915B - Not Quite Latin Square
# https://codeforces.com/problemset/problem/1915/B

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Diccionario de conteo de letras
    conteo = {}

    # Itera sobre las 3 filas del cuadrado latino
    for fila in range(0,3):

        # Input:
        # 2) fila_actual = Fila actual del cuadrado latino
        fila_actual = input()

        # Itera sobre las letras de la fila
        for letra in fila_actual: 

            # Si la letra ya estaba registrada, entonces suma +1 al contador
            if letra in conteo: 
                conteo[letra] += 1

            # Si la letra no estaba registrada, entonces declara el contador igual a 1
            else: 
                conteo[letra] = 1
    
    # Itera sobre el diccionario de contador
    for letra, total in conteo.items():

        # Devuelve la letra que tenga un contador igual a 2
        if total == 2:
            print(letra)

    

File: Problemas/1915B - Not Quite Latin Square/1915B_Not_Quite_Latin_Square.cpp
// 1915B - Not Quite Latin Square 
// https://codeforces.com/problemset/problem/1915/B

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {
        
        // String con todas las letras del cuadrado latino
        string cuadrado_latino = "";

        // Itera sobre las 3 filas del cuadrado latino
        for(int fila = 0; fila < 3; fila++) {

            // Input:
            // 2) fila_actual = Fila actual del cuadrado latino
            string fila_actual;
            cin >> fila_actual;
            
            // Suma las filas del cuadrado latino al string
            cuadrado_latino += fila_actual;
        }

        // Contadores de letras
        int contadorA = 0;
        int contadorB = 0;
        int contadorC = 0;

        // Cuenta el número de instancias de cada letra
        for(int letra = 0; letra < 9; letra++) {
            if(cuadrado_latino[letra] == 'A') {
                contadorA++;
            }
            if(cuadrado_latino[letra] == 'B') {
                contadorB++;
            }
            if(cuadrado_latino[letra] == 'C') {
                contadorC++;
            }
        }

        // Devuelve la letra que tenga un contador igual a 2
        if(contadorA == 2) {
            cout << 'A' << endl;
        }
        if(contadorB == 2) {
            cout << 'B' << endl;
        }
        if(contadorC == 2) {
            cout << 'C' << endl;
        }
    }

    return 0;
}